cmd/metal-api/internal/service/partition-service.go

```diff
@@ -413,6 +414,16 @@ func (r *partitionResource) calcPartitionCapacity(pcr *v1.PartitionCapacityReque
 		return nil, err
 	}
 
+	// if filtered on partition get all without more filters for issues evaluation
+	if machineQuery.PartitionID != nil {
+		err = r.ds.SearchMachines(&datastore.MachineSearchQuery{PartitionID: machineQuery.PartitionID}, &allMs)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		allMs = ms
```

Comment:
`ms` could be filtered by size, so these are not all machines. There should be no other filters applied for machine evaluation (except maybe partition).